numba/core/config.py

```diff
@@ -119,6 +119,20 @@ def update(self, force=False):
             # Store a copy
             self.old_environ = dict(new_environ)
 
+        self.validate()
+
+    def validate(self):
+        if CUDA_USE_NVIDIA_BINDING:  # noqa: F821
+            try:
+                import cuda  # noqa: F401
+            except ImportError as ie:
+                msg = ("CUDA Python bindings requested, "
+                       "but they are not importable")
+                raise RuntimeError(msg) from ie
```

Comment:
This breaks `numba -s` in the following scenario:
* env var set: `NUMBA_CUDA_USE_NVIDIA_BINDING=1`
* `cuda-python` package not installed.

Traceback:
```
$ NUMBA_CUDA_USE_NVIDIA_BINDING=1 numba -s
Traceback (most recent call last):
  File "<pkg_path>/numba/core/config.py", line 127, in validate
    import cuda  # noqa: F401
ModuleNotFoundError: No module named 'cuda'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<env>/bin/numba", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "<pkg_path>/bin/numba", line 5, in <module>
    from numba.misc.numba_entry import main
  File "<pkg_path>/numba/__init__.py", line 19, in <module>
    from numba.core import config
  File "<pkg_path>/numba/core/config.py", line 490, in <module>
    _env_reloader = _EnvReloader()
  File "<pkg_path>/numba/core/config.py", line 85, in __init__
    self.reset()
  File "<pkg_path>/numba/core/config.py", line 89, in reset
    self.update(force=True)
  File "<pkg_path>/numba/core/config.py", line 122, in update
    self.validate()
  File "<pkg_path>/numba/core/config.py", line 131, in validate
    raise RuntimeError(msg) from ie
RuntimeError: CUDA Python bindings requested, but they are not importable
```
I think by extension it would also break the import of Numba itself the same way. e.g. Same set up as above then:
```
python -c 'import numba'
```
produces the same error/behaviour.

Reply:
Yes - I did want to break Numba when that variable is set but not usable, to make sure people didn't think they were running with the bindings when they were not. I think the best of both worlds is:

- `import numba` breaks with that environment variable set but bindings not available
- `numba -s` still works but reports that this problem is present.

Reply:
Thanks for confirming. I think the suggested seems reasonable as it provides a diagnostic method via `numba -s`. Do you think the `RuntimeError` message ought to explain why the "CUDA Python bindings requested" happened, maybe:
```
CUDA Python bindings requested (environment variable NUMBA_CUDA_USE_NVIDIA_BINDING is set), but they are not importable
```
so as to hint to users where to start looking?

Reply:
From OOB conversation, decided to make this a warning as it's possible users e.g. reuse shell inits with this set but, for example, just want to use the CPU target.